packages/devtools_app/lib/src/performance/legacy/performance_model.dart

```diff
@@ -127,9 +127,17 @@ class LegacyPerformanceData {
   static int eventGroupComparator(String a, String b) {
     if (a == b) return 0;
 
-    // Order Unknown buckets last.
-    if (a == unknownKey) return 1;
-    if (b == unknownKey) return -1;
+    // TODO(kenz): Once https://github.com/flutter/flutter/issues/83835 is
+    // addressed, match on the group key that all skia.shader events will have.
+    // Order shader buckets first.
+    if (a.toLowerCase().contains('shade')) return -1;
```

Comment:
This isn't a valid comparator.
```dart
compare('shade A', 'shade B') == -1
compare('shade B', 'shade A') == -1
compare('shade A', 'shade A') == -1
```dart
This will cause sort to do funky things
What you want is
```dart
final shadeCompare = a.contains('shade').compareTo(b.contains('shade'));
if (shadeCompare != 0) return -shadeCompare; // because you want it first.
```
The same would also simplify the logic for the unknownKey logic.

Reply:
fixed and added more tests.